### PR TITLE
Check if public priorities and predicates are registered or used

### LINKED ISSUE #21236: Progamatically ensure that all predicate/priority functions that are tested are registered
@resouer I just noticed you forgot to register the image locality priority function in `init()` in `plugin/pkg/scheduler/algorithmprovider/defaults/defaults.go`

I also missed it when I reviewed your PR.

I have fixed this in #21235. But it would be great if you could figure out a mechanism that would ensure that all predicate/priority functions that are tested (you can assume they will always live under `plugin/pkg/scheduler/algorithm/{predicates,priorities}`) are registered, and fail the test if not. That will allow us to prevent this mistake in the future.

### PR DESCRIPTION
**Pull request by resouer**
fix #21236

We need verification to make sure all implemented public priorities and predicates under `plugin/pkg/scheduler/algorithm/{predicates,priorities}` are registered or used in other places.

cc @davidopp

### CODE REVIEW

**Comment by resouer** [comment]
ping @davidopp

**Comment by brendandburns** [comment]
Thanks!

**Comment by davidopp** [comment]
cc/ @HaiyangDING

File: pkg/util/codeinspector/codeinspector.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"io/ioutil"
+	"strings"
+	"unicode"
+)
+
+// Get all public functions (not methods) from a golang source file.
```
Comment (HaiyangDING) [inline pkg/util/codeinspector/codeinspector.go]: Comments for any function should start with the function name, i.e. "GetPublicFunctions gets all public functions (not methods) from a golang source file."

File: pkg/util/codeinspector/codeinspector.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+func GetPulicFunctions(filePath string) ([]string, error) {
```
Comment (HaiyangDING) [inline pkg/util/codeinspector/codeinspector.go]: s/GetPulicFunctions/GetPublicFunctions

File: pkg/util/codeinspector/codeinspector.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+			}
+		}
+		return true
+	})
+
+	return functionNames, nil
+}
+
+// Check if a given string is a public function name.
+func isPublic(myString string) bool {
```
Comment (HaiyangDING) [inline pkg/util/codeinspector/codeinspector.go]: Comments for any function should start with the function name.

File: pkg/util/codeinspector/codeinspector.go [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+	a := []rune(myString)
+	a[0] = unicode.ToUpper(a[0])
+	return myString == string(a)
+}
+
+// List golang source code files from directory, excluding sub-directory and tests files.
+func GetSourceCodeFiles(dir string) ([]string, error) {
```
Comment (HaiyangDING) [inline pkg/util/codeinspector/codeinspector.go]: Comments for any function should start with the function name.

File: plugin/pkg/scheduler/algorithm/priorities/priorities_test.go [hunk 5]
```diff
@@ -884,3 +886,47 @@ func makeImageNode(node string, status api.NodeStatus) api.Node {
 		Status:     status,
 	}
 }
+
+func TestPrioritiesRegistered(t *testing.T) {
+	var functionNames []string
+
+	// Files and directories which predicates may be referenced
```
Comment (HaiyangDING) [inline plugin/pkg/scheduler/algorithm/priorities/priorities_test.go]: s/predicates/priorities

File: plugin/pkg/scheduler/algorithm/predicates/predicates_test.go [hunk 6]
```diff
@@ -1408,3 +1410,48 @@ func TestEBSVolumeCountConflicts(t *testing.T) {
...
+	}
+
+	// List all golang source files under ./predicates/, excluding test files and sub-directories.
+	files, err := codeinspector.GetSourceCodeFiles(".")
+
+	if err != nil {
+		t.Errorf("unexpected error: %v when listing files in current directory", err)
+	}
+
+	// Get all public predicates in priority files.
```
Comment (HaiyangDING) [inline plugin/pkg/scheduler/algorithm/predicates/predicates_test.go]: s/priority/predicate

or simply "Get all public predicates" would be enough, IMO.

**Comment by resouer** [comment]
updated the comments, thanks

**Pull request merged**